builder.go

```diff
@@ -0,0 +1,99 @@
+package main
+
+import (
+	"errors"
+	"log"
+
+	"github.com/hashicorp/packer/common"
+	"github.com/hashicorp/packer/packer"
+	"github.com/mitchellh/multistep"
+	"github.com/hashicorp/packer/helper/communicator"
+	gossh "golang.org/x/crypto/ssh"
+	"github.com/hashicorp/packer/communicator/ssh"
+)
+
+type Builder struct {
+	config *Config
+	runner multistep.Runner
+}
+
+func (b *Builder) Prepare(raws ...interface{}) ([]string, error) {
+	c, warnings, errs := NewConfig(raws...)
+	if errs != nil {
+		return warnings, errs
+	}
+	b.config = c
+
+	return warnings, nil
+}
+
+func (b *Builder) Run(ui packer.Ui, hook packer.Hook, cache packer.Cache) (packer.Artifact, error) {
+	// Set up the state.
+	state := new(multistep.BasicStateBag)
+	state.Put("hook", hook)
+	state.Put("ui", ui)
+
+	// Build the steps.
+	steps := []multistep.Step{
+		&StepConfigureHW{
```

Comment:
Okay. But what is the difference between this approach and creating a VM with all the settings already set?